yagmail/message.py

```diff
@@ -152,7 +153,13 @@ def prepare_contents(contents, encoding):
     mime_objects = []
     has_included_images = False
     if contents is not None:
+        unnamed_attachment_id = 1
         for is_marked_up, content in contents:
+            if isinstance(content, io.IOBase):
+                if not hasattr(content, 'name'):
+                    # If the IO object has no name attribute, give it one.
+                    content.name = f'attachment_{unnamed_attachment_id}'
```

Comment:
Maybe there is a way we can guess extension too based on the first few bytes... not sure if that's how it could work. In any case, I am pretty pleased that very often this name attr would already be set.

Reply:
There are libraries for that! e.g. [python-magic](https://github.com/ahupp/python-magic). If you are willing to rely on a dependency this seems to be an easy solution.

e.g.
```python
>>> magic.from_buffer(content.read(2048), mime=True)
'application/pdf'
>>> suffix = magic.from_buffer(content.read(2048), mime=True).split('/')[1]
>>> content.name = f'attachment_{unnamed_attachment_id}.{suffix}' 
```

Reply:
I was excited about this, but I suspect this won't work so easily on Windows.

Reply:
I think we'll just have to assume that the name attribute will be set. The documentation should probably very much hint at how to do this, though realistically I guess it will be mostly set, so well done!